Userland/Applications/MailSettings/MailSettingsWindow.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2021, The SerenityOS developers
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "MailSettingsWindow.h"
+#include <Applications/MailSettings/MailSettingsWindowGML.h>
+#include <LibCore/ConfigFile.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/CheckBox.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/SpinBox.h>
+#include <LibGUI/TabWidget.h>
+#include <LibGUI/TextBox.h>
+#include <LibGUI/Widget.h>
+#include <unistd.h>
+
+void MailSettingsWindow::reset_default_values()
+{
+    m_server_inputbox->set_text(m_server);
+    m_port_spinbox->set_value(m_port);
+    m_tls_checkbox->set_checked(m_tls);
+    m_email_inputbox->set_text(m_email);
+    update();
+}
+
+void MailSettingsWindow::write_values()
+{
+    m_server = m_server_inputbox->get_text();
+    m_port = m_port_spinbox->value();
+    m_tls = m_tls_checkbox->is_checked();
+    m_email = m_email_inputbox->get_text();
+
+    m_config->write_entry("Connection", "Server", m_server);
+    m_config->write_num_entry("Connection", "Port", m_port);
+    m_config->write_bool_entry("Connection", "TLS", m_tls);
+    m_config->write_entry("User", "Username", m_email);
+    m_config->sync();
+}
+
+MailSettingsWindow::MailSettingsWindow()
+{
+    m_config = Core::ConfigFile::get_for_app("Mail");
+    if (unveil(m_config->filename().characters(), "rwc") < 0) {
+        perror("unveil");
+        GUI::Application::the()->quit();
+    }
+
+    if (unveil("/res", "r") < 0) {
+        perror("unveil");
+        GUI::Application::the()->quit();
+    }
```

Comment:
You're not closing the veil here (i.e. by calling unveil with nullptr for both arguments) which means those unveil calls have no effect. Also, consider moving those into the main function for consistency with other applications.